boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -140,31 +139,29 @@ mb1:
 
 mb2:
     /* See if the boot device was passed in */
-    test dword ptr ds:[ebx + MB_INFO_FLAGS_OFFSET], MB_INFO_FLAG_BOOT_DEVICE
+    test dword ptr [rbx + MB_INFO_FLAGS_OFFSET], MB_INFO_FLAG_BOOT_DEVICE
 
     /* If no boot device known, assume first partition of first harddisk */
     mov dx, HEX(0180)
     jz mb3
 
     /* Load boot drive into DL, boot partition into DH */
-    mov edx, dword ptr ds:[ebx + MB_INFO_BOOT_DEVICE_OFFSET]
+    mov rdx, [rbx + MB_INFO_BOOT_DEVICE_OFFSET]
     bswap edx
-    inc dh
 
 mb3:
     /* Relocate itself to lower address */
     mov esi, INITIAL_BASE
     mov edi, FREELDR_BASE
-    mov ecx, offset __bss_start__ - FREELDR_BASE
-    shr ecx, 2
-    rep movsd
+    mov ecx, offset __bss_end__ - FREELDR_BASE
```

Comment:
Is this change of bss_start to bss_end to ensure that the (in principle, already zeroed-out) bss part also gets copied?

Reply:
No. Sorry i will put back the `__bss_start__`